fpm/src/fpm_backend.f90

```diff
@@ -32,127 +32,102 @@ subroutine build_package(model)
         call mkdir(join_path(model%output_directory,model%package_name))
     end if
 
-    linking = ""
-    do i=1,size(model%sources)
+    if (model%targets(1)%ptr%target_type == FPM_TARGET_ARCHIVE) then
+        linking = ' -l'//model%package_name//" -L"//&
+                    join_path(model%output_directory,model%package_name)
```

Comment:
Should we use the absolute path to the archive here instead? Currently this will fail if I name my package `m`:
```
/usr/bin/ld: build/gfortran_release/m/libm.a(fpm_filesystem.f90.o): in function `__fpm_filesystem_MOD_get_temp_filename':
fpm_filesystem.f90:(.text+0xce): warning: the use of `tempnam' is dangerous, better use `mkstemp'
/usr/bin/ld: build/gfortran_release/m/libm.a(dependencies_M_CLI2_src_M_CLI2.f90.o): undefined reference to symbol 'logf@@GLIBC_2.27'
/usr/bin/ld: /usr/lib/../lib/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
 Command failed
```